### ISSUE TITLE
[BDSP] Stationary Gift Generation (Jirachi) - Incorrect Natures

### ISSUE THREAD

**Issue by zwab** [Admiral-Fish/PokeFinder#237]
Unsure if this has been flagged already, but while testing some stuff on the latest version of the game, I note the nature shown for gift pokemon is incorrect in release 4.0.0 (Shows Sassy instead of Lonely):

![RNG Reporter Release v4 0 0](https://user-images.githubusercontent.com/51080317/155137878-5b6de553-ac00-4d85-b2f4-d9004839fc55.png)

This issue exists in the latest appveyor commit (c1c70e6f1650b4070bc44ee02b72b31ef73f66db)

![RNG Reporter Commit c1c70e6f1650b4070bc44ee02b72b31ef73f66db](https://user-images.githubusercontent.com/51080317/155137989-2b309826-9534-4aa7-b1a8-2253b83a0a13.png)

I've inclued my seed info below:

```
d65ea9071d483605
d7a0c4b398b5c671

Target: 7438
```

Also included output from the bot I was testing, which includes the advance hit + PID/EC/Nature/IVs of generated Jirachi.

```
Advances: 7361


Current Seed
S[0]: 6CEE6CBA  S[1]: 19C67DB0
S[2]: 34371311  S[3]: 461E0386

Advances: 7393


Current Seed
S[0]: 34371311  S[1]: 461E0386
S[2]: 598A3C43  S[3]: 738BE13F

Advances: 7395


Current Seed
S[0]: 598A3C43  S[1]: 738BE13F
S[2]: FFA8DBC4  S[3]: 491CF584

Advances: 7397


Current Seed
S[0]: 6DD223C9  S[1]: D41DA34C
S[2]: 7A03DC9E  S[3]: DE5D99AE

Advances: 7402


Current Seed
S[0]: 24133F1E  S[1]: C07AC423
S[2]: BAA114EB  S[3]: AD624124

Advances: 7434


Current Seed
S[0]: BAA114EB  S[1]: AD624124
S[2]: 10347059  S[3]: 0679F5A0

Advances: 7436


Current Seed
S[0]: 10347059  S[1]: 0679F5A0
S[2]: B4CDBFC8  S[3]: 0B19A314

Advances: 7438


We hit 7438!


Exiting...
Bot Disconnected
PS D:\Python\Python3\PyNXBot\PyNXBot-master_fork_commit_ac2f7c752cd5d3e5deced88ae3f368faff685fee\PyNXBot-master\tests> python3 .\BDSP-CheckParty.py
Bot Connected
Game: Shining Pearl    Version: 1.2.0
G8TID: 167938    TID: 60482    SID: 27971

Slot: 1
EC: 86252EFC  PID: 4D8CCD9C  Ponyta
Nature: Bold(Bold)  Ability: Flame Body(H)  Gender: ♀
IVs: [18, 7, 22, 17, 8, 0]  EVs: [5, 2, 9, 4, 4, 1]
Moves: Stomp / Flame Wheel / Tail Whip / Ember

Slot: 2
EC: 95E6A26C  PID: C5FC44FC  ⋆ Empoleon
Nature: Calm(Calm)  Ability: Torrent(2)  Gender: ♀
IVs: [31, 15, 28, 21, 2, 16]  EVs: [49, 99, 81, 91, 37, 153]
Moves: Flash Cannon / Drill Peck / Hydro Pump / Surf

Slot: 3
EC: 381CAC96  PID: 28EB58D7  Jirachi
Nature: Lonely(Lonely)  Ability: Serene Grace(2)  Gender: -
IVs: [31, 10, 31, 31, 31, 31]  EVs: [0, 0, 0, 0, 0, 0]
Moves: Confusion / Wish / ——— / ———

Slot: 4
Empty
Slot: 5
Empty
Slot: 6
Empty

Exiting...
Bot Disconnected
PS D:\Python\Python3\PyNXBot\PyNXBot-master_fork_commit_ac2f7c752cd5d3e5deced88ae3f368faff685fee\PyNXBot-master\tests>
```

**Comment by Admiral-Fish** [comment]
https://github.com/Admiral-Fish/PokeFinder/pull/219 potential cause

**Comment by SteveCookTU** [comment]
@zwab Do you possibly have a Let's Go save to test mew as well? There is a chance it may also be affected but I am unable to confirm myself.

**Comment by zwab** [comment]
> @zwab Do you possibly have a Let's Go save to test mew as well? There is a chance it may also be affected but I am unable to confirm myself.

I do indeed, results below:


Pokefinder release 4.0.0 shows Lonely nature for advance 12404, actual Mew was Bold:

![image](https://user-images.githubusercontent.com/51080317/155278436-c75afe8b-41f8-446b-aab6-8080a7b4b171.png)


Latest AppVeyor commit, f72c7fe2efab2c0ed1822db8965d896ecbb00c5d, also shows lonely nature:

![image](https://user-images.githubusercontent.com/51080317/155278960-0bb9068f-21cf-4a82-89f2-c0379dce6c61.png)




Seed Info:

```
e7e9d5fe5ecf461d
e2fbb021cd3d6d01

Target: 12404
```


Info from bot, including party members:

```
Advances: 12322


Current Seed
S[0]: FAAF7087  S[1]: FD2E17C4
S[2]: 30A00831  S[3]: C8DDEEEC

Advances: 12351


Current Seed
S[0]: FD2E17C4  S[1]: 30A00831
S[2]: C8DDEEEC  S[3]: 49779438

Advances: 12352


Current Seed
S[0]: 30A00831  S[1]: C8DDEEEC
S[2]: 49779438  S[3]: C46A3AE5

Advances: 12353


Current Seed
S[0]: C46A3AE5  S[1]: F4BB43D9
S[2]: D336792C  S[3]: 2615E126

Advances: 12356


Current Seed
S[0]: D336792C  S[1]: 2615E126
S[2]: B33D4A13  S[3]: 9DB67226

Advances: 12358


Current Seed
S[0]: 74BC55F7  S[1]: 2D88E4A1
S[2]: 03818474  S[3]: B2243F0E

Advances: 12363


Current Seed
S[0]: D60FC685  S[1]: EA32DAE3
S[2]: 4920DCF8  S[3]: 942FC5EA

Advances: 12393


Current Seed
S[0]: EA32DAE3  S[1]: 4920DCF8
S[2]: 942FC5EA  S[3]: 3CBC0204

Advances: 12394


Current Seed
S[0]: 3CBC0204  S[1]: 402522B2
S[2]: 0FADF152  S[3]: E54765D8

Advances: 12397


Current Seed
S[2]: 3937F756  S[3]: 506E7270

Advances: 12399


Current Seed
S[0]: EE9EF436  S[1]: 689523F4
S[2]: 4B4B2164  S[3]: 32E9BD0D

Advances: 12404


We hit 12404!


Exiting...
Bot Disconnected
PS D:\Python\Python3\PyNXBot\PyNXBot-master_fork_commit_ac2f7c752cd5d3e5deced88ae3f368faff685fee\PyNXBot-master\tests> python3 .\BDSP-CheckParty.py
Bot Connected
Game: Shining Pearl    Version: 1.2.0
G8TID: 167938    TID: 60482    SID: 27971

Slot: 1
EC: 86252EFC  PID: 4D8CCD9C  Ponyta
Nature: Bold(Bold)  Ability: Flame Body(H)  Gender: ♀
IVs: [18, 7, 22, 17, 8, 0]  EVs: [5, 2, 9, 4, 7, 1]
Moves: Stomp / Flame Wheel / Tail Whip / Ember

Slot: 2
EC: ABCFC022  PID: 78D69019  Mew
Nature: Bold(Bold)  Ability: Synchronize(2)  Gender: -
IVs: [31, 0, 31, 31, 31, 31]  EVs: [0, 0, 0, 0, 0, 0]
Moves: Pound / Reflect Type / ——— / ———

Slot: 3
Empty
Slot: 4
Empty
Slot: 5
Empty
Slot: 6
Empty

Exiting...
Bot Disconnected
```

**Comment by Real96** [comment]
Fixed in 14e11b8
